server/datastore/mysql/apple_mdm.go

```diff
@@ -1732,6 +1672,43 @@ WHERE
 	return nil
 }
 
+func (ds *Datastore) GetHostMDMProfilesExpectedForVerification(ctx context.Context, host *fleet.Host) (map[string]*fleet.ExpectedMDMProfile, error) {
+	var teamID uint
+	if host.TeamID != nil {
+		teamID = *host.TeamID
+	}
+
+	stmt := `
+SELECT
+	identifier,
+	earliest_install_date
+FROM
+	mdm_apple_configuration_profiles macp
+	JOIN (
+		SELECT
+			checksum,
+			min(updated_at) AS earliest_install_date
+		FROM
+			mdm_apple_configuration_profiles
+		GROUP BY
+			checksum) cs 
+	ON macp.checksum = cs.checksum
+WHERE
+	macp.team_id = ?`
```

Comment:
are you doing the `JOIN` to grab the identifier? keep in mind that the identifier is part of the XML => part of the checksum, so if they have the same checksum, they have the same identifier as well.

Reply:
Yes, the `JOIN` is so we can use the results of this query to match against the osquery results (which don't have a checksum). I suppose we could alternatively do some kind of SQL aggregate function like `min(identifier)` in order to `GROUP BY`, but that feels a bit strange. Originally, I was also grabbing the `updated_at` time for the expected team profile and another column for the `earliest_install_date`, which is how I landed on the `JOIN` approach to preserve the team association.

Reply:
I think I was missing the fact that there's no way around the `WHERE map.team_id = ?`